File: s04_CatchSpider/CatchSpider/app/src/main/java/com/omeryildizce/catchspider/MainActivity.java
package com.omeryildizce.catchspider;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.os.Handler;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    TextView timeText;
    TextView scoreText;
    int score;
    ImageView imageView1;
    ImageView imageView2;
    ImageView imageView3;
    ImageView imageView4;
    ImageView imageView5;
    ImageView imageView6;
    ImageView imageView7;
    ImageView imageView8;
    ImageView imageView9;
    ImageView[] imageArrays;
    Handler handler;
    Runnable runnable;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        timeText = findViewById(R.id.timeText);
        scoreText = findViewById(R.id.scoreText);
        imageView1 = findViewById(R.id.imageView1);
        imageView2 = findViewById(R.id.imageView2);
        imageView3 = findViewById(R.id.imageView3);
        imageView4 = findViewById(R.id.imageView4);
        imageView5 = findViewById(R.id.imageView5);
        imageView6 = findViewById(R.id.imageView6);
        imageView7 = findViewById(R.id.imageView7);
        imageView8 = findViewById(R.id.imageView8);
        imageView9 = findViewById(R.id.imageView9);
        imageArrays = new ImageView[]{
                imageView1, imageView2, imageView3, imageView4, imageView5, imageView6, imageView7, imageView8, imageView9
        };

        hideImages();
        score = 0;
        new CountDownTimer(10_000, 1_000) {
            @Override
            public void onTick(long millisUntilFinished) {
                timeText.setText("Time: " + (millisUntilFinished / 1_000));
            }

            @Override
            public void onFinish() {
                timeText.setText("Time Off");
                handler.removeCallbacks(runnable);
                for (ImageView image : imageArrays) {
                    image.setVisibility(View.INVISIBLE);
                }

                AlertDialog.Builder alert = new AlertDialog.Builder(MainActivity.this);
                alert.setTitle("Restart?");
                alert.setMessage("Are you sure to restart game? ");
                alert.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        Intent intent = getIntent();
                        finish();
                        startActivity(intent);
                    }
                });
                alert.setNegativeButton("No", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        Toast.makeText(MainActivity.this, "Score: " + score+"\nGame Over!", Toast.LENGTH_SHORT).show();
                    }
                });

                alert.show();
            }
        }.start();
    }

    public void increaseScore(View view) {
        score++;
        scoreText.setText(getResources().getString(R.string.score_00) + " " + score);
    }

    public void hideImages() {
        handler = new Handler();
        runnable = new Runnable() {
            @Override
            public void run() {
                for (ImageView image : imageArrays) {
                    image.setVisibility(View.INVISIBLE);
                }
                Random random = new Random();
                int index = random.nextInt(9);
                imageArrays[index].setVisibility(View.VISIBLE);
                handler.postDelayed(this, 500);
            }
        };
        handler.post(runnable);
    }
}

File: s17_CatchTheCat/CatchTheCat/app/src/main/java/com/omeryildizce/catchthecat/MainActivity.kt
package com.omeryildizce.catchthecat

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import android.os.Handler
import android.view.View
import android.widget.ImageView
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import com.omeryildizce.catchthecat.databinding.ActivityMainBinding
import java.util.*
import kotlin.collections.ArrayList

class MainActivity : AppCompatActivity() {
    private var score = 0
    private lateinit var binding: ActivityMainBinding
    var imageArray = ArrayList<ImageView>()
    var runnable = Runnable { }
    var handler = Handler()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        // Image Array
        initialize()
        hideImages()
        showImages()


        object : CountDownTimer(15_000, 1000) {
            override fun onTick(millisUntilFinished: Long) {
                binding.time.text = String.format("Time: 00:%02d", millisUntilFinished / 1000)
            }

            override fun onFinish() {
                binding.time.text = String.format("Time: 00:00")
                handler.removeCallbacks(runnable)
                hideImages()
                val alert = AlertDialog.Builder(this@MainActivity)
                alert.setTitle("Game Over")
                alert.setMessage("Your Score $score\nRestart The Game?")
                alert.setPositiveButton("yes") { dialog, which ->
                    val intent = intent
                    finish()
                    startActivity(intent)
                }
                alert.setNegativeButton("no") { dialog, which ->
                    Toast.makeText(this@MainActivity, "Game Over", Toast.LENGTH_LONG).show()
                }

                alert.show()
            }
        }.start()
    }

    private fun showImages() {
        runnable = object : Runnable {
            override fun run() {
                hideImages()
                val random = Random()
                val randomNumber = random.nextInt(30)
                imageArray[randomNumber].visibility = View.VISIBLE
                handler.postDelayed(runnable,750)

            }
        }

        handler.post(runnable)

    }
    private fun hideImages(){
        for (image in imageArray) {
            image.visibility = View.INVISIBLE
        }
    }
    private fun initialize() {
        imageArray.add(binding.cat01)
        imageArray.add(binding.cat02)
        imageArray.add(binding.cat03)
        imageArray.add(binding.cat04)
        imageArray.add(binding.cat05)
        imageArray.add(binding.cat06)
        imageArray.add(binding.cat07)
        imageArray.add(binding.cat08)
        imageArray.add(binding.cat09)
        imageArray.add(binding.cat10)

        imageArray.add(binding.cat11)
        imageArray.add(binding.cat12)
        imageArray.add(binding.cat13)
        imageArray.add(binding.cat14)
        imageArray.add(binding.cat15)
        imageArray.add(binding.cat16)
        imageArray.add(binding.cat17)
        imageArray.add(binding.cat18)
        imageArray.add(binding.cat19)
        imageArray.add(binding.cat20)

        imageArray.add(binding.cat21)
        imageArray.add(binding.cat22)
        imageArray.add(binding.cat23)
        imageArray.add(binding.cat24)
        imageArray.add(binding.cat25)
        imageArray.add(binding.cat26)
        imageArray.add(binding.cat27)
        imageArray.add(binding.cat28)
        imageArray.add(binding.cat29)
        imageArray.add(binding.cat30)
    }

    fun increaseScore(view: View) {
        score++
        binding.score.text = String.format("Score: %02d", score)
    }

}